source/precharge/tmu/tmu.c

```diff
@@ -0,0 +1,30 @@
+#include "tmu.h"
+
+void initTMU(tmu_handle_t *tmu) {
+    uint8_t spi_tx_buffer[3] = {0};
+    uint8_t spi_rx_buffer[3] = {0};
+    while(PHAL_SPI_busy())
+        ;
+    spi_tx_buffer[0] = TMU_PRODID_ADDR;
+    PHAL_SPI_transfer(tmu->spi, spi_tx_buffer, 3, spi_rx_buffer);
```

Comment:
After you do the transfer, check the values in the receive buffer are as expected. If not, return false. If its good, return true.

Reply:
Your goal in this init function is to check the chip is present and is working properly.